### PR TITLE
builder: Enable building images on arm64

### LINKED ISSUE #17056: Enable Cilium images to be built from ARM instance
## Bug report

**General Information**

Cilium image builds fine on x86s build platform (including ARM cross compilation from x86s). However if ARM instance is used as build platform to build ARM target, compilation fails. 

- Cilium version (run `cilium version`)
- Kernel version (run `uname -a`)
```
uname -a
Linux ip-x-x-x-x.us-west-2.compute.internal 4.14.238-182.422.amzn2.aarch64 #1 SMP Tue Jul 20 20:36:04 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux
```

- Orchestration system version in use (e.g. `kubectl version`, ...)  NA
- Link to relevant artifacts (policies, deployments scripts, ...) NA
- Generate and upload a system zip:  NA
```
curl -sLO https://git.io/cilium-sysdump-latest.zip && python cilium-sysdump-latest.zip
```

**How to reproduce the issue**

* Start a new m6g.4xlarge instance running AL2. Launch instance with 256 Gi EBS volume to avoid below build error (error: failed to solve: failed to copy files: copy file range failed: no space left on device)
* Install docker - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/docker-basics.html
* Install docker buildx (https://www.docker.com/blog/getting-started-with-docker-for-arm-on-linux/)
* Get cilium source code (git clone https://github.com/cilium/cilium.git)
* ARCH=aarch64 DOCKER_IMAGE_TAG=test-1 make docker-images-all

Build would fail as "binutils-x86-64-linux-gnu" is not installed on container.  Compilation error points to images/cilium/download-hubble.sh  
```
x86_64-linux-gnu-strip /out/linux/amd64/bin/hubble. (<<<---- Fails here. ---->>>)
aarch64-linux-gnu-strip /out/linux/arm64/bin/hubble
```

Fixed build error by installing the required binary 

```
[ec2-user@ip cilium]$ git diff images/cilium/download-hubble.sh 
[ec2-user@ip cilium]$ git diff
diff --git a/images/cilium/Dockerfile b/images/cilium/Dockerfile
index 16600e68d..3f509f362 100644
--- a/images/cilium/Dockerfile
+++ b/images/cilium/Dockerfile
@@ -16,8 +16,10 @@ FROM quay.io/cilium/cilium-envoy:e83275091139101e7ff29284ef35c56f4cd6bfc3@sha256
 FROM --platform=${BUILDPLATFORM} ${CILIUM_BUILDER_IMAGE} as hubble
 COPY images/cilium/hubble-version.sh /tmp/hubble-version.sh
 COPY images/cilium/download-hubble.sh /tmp/download-hubble.sh
+RUN apt-get update && apt-get install -y binutils-aarch64-linux-gnu
+RUN apt-get update && apt-get install -y binutils-x86-64-linux-gnu
 RUN /tmp/download-hubble.sh
```

Similar changes are required in below files
```
	modified:   images/clustermesh-apiserver/Dockerfile
	modified:   images/hubble-relay/Dockerfile
	modified:   images/operator/Dockerfile
	modified:   images/runtime/Dockerfile
```
Happy to submit a PR for the above changes.

### PR DESCRIPTION
**Pull request by jrajahalme**
Add cross-compilation packages to builder when build target is
arm64. This allows Cilium image build to succeed. Without this
builer/cilium/download-hubble.sh failed due to missing
'x86_64-linux-gnu-strip'.

Also install 'binutils-x86-64-linux-gnu' whenever installing
'binutils-aarch64-linux-gnu' to fix gops build on arm64.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

```release-note
Cilium images can now be built also on arm64.
```

Note: This PR supersedes #17948 in order to follow through with https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images.

Fixes: #17056

### CODE REVIEW

**Comment by jrajahalme** [comment]
@joestringer Keeping this as draft until I get through all the hoop in  https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images

**Comment by jrajahalme** [comment]
@joestringer runtime image build is waiting for review, status shows as:
```
Waiting for review: release-base-images needs approval to start deploying changes.
```

I did not see this step being documented, unless this is covered by:

> Ping one of the members of team/build to approve the build that was created by GitHub Actions here. Note that at this step cilium-builder build failure is expected since we have yet to update the runtime digest.

Maybe review == approval here?

**Comment by jrajahalme** [comment]
Opened packer-ci-build PR to update to the new builder image: https://github.com/cilium/packer-ci-build/pull/299

**Comment by jrajahalme** [comment]
Travis CI build on amd64 failed due to docker pull rate limit.

**Comment by jrajahalme** [comment]
net-next failed with #17604

**Comment by jrajahalme** [comment]
4.19 fail also looks unrelated to this PR.

**Comment by jrajahalme** [comment]
Added a new commit to bump default vagrant box versions to those built from https://github.com/cilium/packer-ci-build/pull/299.

The default box versions seemed to be somewhat out ot date?

**Comment by tklauser** [comment]
> Added a new commit to bump default vagrant box versions to those built from [cilium/packer-ci-build#299](https://github.com/cilium/packer-ci-build/pull/299).
> 
> The default box versions seemed to be somewhat out ot date?

The box versions specified in commit ffcb37a refer to the box versions of the PR runs, not the master runs. We need to trigger the vagrant boxes to be built from master and then update to these versions.

**Comment by jrajahalme** [comment]
> > Added a new commit to bump default vagrant box versions to those built from [cilium/packer-ci-build#299](https://github.com/cilium/packer-ci-build/pull/299).
> > The default box versions seemed to be somewhat out ot date?
> 
> The box versions specified in commit [ffcb37a](https://github.com/cilium/cilium/commit/ffcb37ae9f53d439d93633f3c69059a3f01ebe45) refer to the box versions of the PR runs, not the master runs. We need to trigger the vagrant boxes to be built from master and then update to these versions.

Updated to the new packer-ci-build master builds.

**Review by tklauser — APPROVED** [review]

File: vagrant_box_defaults.rb [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
+$NETNEXT_SERVER_VERSION= "114"
 @v54_SERVER_BOX= "cilium/ubuntu-5-4"
-@v54_SERVER_VERSION= "12"
+@v54_SERVER_VERSION= "19"
 @v419_SERVER_BOX= "cilium/ubuntu-4-19"
-@v419_SERVER_VERSION= "48"
+@v419_SERVER_VERSION= "53"
 @v49_SERVER_BOX= "cilium/ubuntu"
-@v49_SERVER_VERSION= "208"
+@v49_SERVER_VERSION= "213"
```
Comment (pchaigno) [inline vagrant_box_defaults.rb]: These updates need to be in a PR with a specific name. A PR is already opened at https://github.com/cilium/cilium/pull/17761 for that, but a couple CI tests are consistently failing. I'm happy to update the box versions in that PR once I've debugged CI.

**Review by pchaigno — CHANGES_REQUESTED** [review]

File: vagrant_box_defaults.rb (hunk 1 above)
Comment (jrajahalme) [reply vagrant_box_defaults.rb]: Did not know, dropped the last commit :-)

**Review by pchaigno — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Comment by tklauser** [comment]
This PR picked up a merge conflict and now needs a rebase.

**Review by pchaigno — APPROVED** [review]

**Comment by pchaigno** [comment]
The GKE CI job failed with known flake https://github.com/cilium/cilium/issues/17307; k8s-1.23-kernel-net-next with known flake https://github.com/cilium/cilium/issues/18447. Reviews are in. Merging.

**Pull request merged**